### PR TITLE
Fix/bitcoin por list protocol

### PR DESCRIPTION
**Pull request by KuphJr**
## Changes

- Fixed bug in the `proof-of-reserves` adapter that caused it to not work using the list protocol
- Added integration tests

## Steps to Test

`yarn test proof-of-reserves`

## Quality Assurance

- [x] Ran `yarn changeset` if adapter source code was changed
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

File: packages/composites/proof-of-reserves/test/integration/adapter.test.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { mockPoRindexerSuccess, mockEthBalanceSuccess } from './fixtures'
+import * as nock from 'nock'
+import { AddressInfo } from 'net'
+
+beforeAll(() => {
+  if (process.env.RECORD) {
+    nock.recorder.rec()
+  }
+  process.env.ETH_BALANCE_ADAPTER_URL = 'http://localhost:8081/'
+  process.env.POR_INDEXER_ADAPTER_URL = 'https://adapters.main.prod.cldev.sh/por-indexer'
```
Comment (austinborn) [inline packages/composites/proof-of-reserves/test/integration/adapter.test.ts:14]: Can you change this to a fake url?

File: packages/composites/proof-of-reserves/src/utils/protocol.ts [hunk 2]
```diff
@@ -41,11 +45,36 @@ export const runProtocolAdapter = async (
   return callAdapter(execute, context, next, '_onProtocol')
 }
 
-const listAdapter = (jobRunID: string, data: { addresses: string[] }) => {
+const listAdapter = (
+  jobRunID: string,
+  data: { addresses: AddressList; chainId?: string; network?: string },
+) => {
+  console.log('LIST ADAPTER')
+  console.log(data)
```
Comment (austinborn) [inline packages/composites/proof-of-reserves/src/utils/protocol.ts:53]: Remove plz

File: packages/composites/proof-of-reserves/src/utils/protocol.ts [hunk 3]
```diff
@@ -41,11 +45,36 @@ export const runProtocolAdapter = async (
...
   if (!('addresses' in data)) {
     throw Error(`Missing "addresses" in request data`)
   }
-
-  const result = data.addresses.map((address: string) => ({ address }))
-  return Requester.success(jobRunID, { data: { result } })
+  if (typeof data.addresses[0] === 'string') {
+    const result = (data.addresses as string[]).map((address: string) => {
+      return { address, network: data.network, chainId: data.chainId }
+    })
```
Comment (austinborn) [inline packages/composites/proof-of-reserves/src/utils/protocol.ts:60]: Nbd but I like to remove the `return` keyword when possible
```suggestion
    const result = (data.addresses as string[]).map((address: string) => (
      { address, network: data.network, chainId: data.chainId }
    ))
```

File: packages/composites/proof-of-reserves/test/integration/adapter.DISABLEDtest.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import * as process from 'process'
```
Comment (justinkaseman) [inline packages/composites/proof-of-reserves/test/integration/adapter.DISABLEDtest.ts:1]: Based on this file's name is it not intended to run?

**Review by boxhock — CHANGES_REQUESTED** [review]

File: packages/composites/proof-of-reserves/src/utils/protocol.ts [hunk 5]
```diff
@@ -23,15 +23,27 @@ export const adapters: AdapterImplementation[] = [
 
 export type Protocol = typeof adapters[number]['NAME']
 
+type AddressData =
+  | { token: string; chainId: string; network: string }
+  | { addresses: AddressList; chainId?: string; network?: string }
+
+type AddressList = string[] | AddressObject[]
+
+type AddressObject = { address: string; network?: string; chainId?: string }
```
[the remark this answers is not in the document]
Comment (KuphJr) [reply packages/composites/proof-of-reserves/src/utils/protocol.ts:32]: The problem is, sometimes they ARE optional.  For example, the 'eth-balance' indexer does NOT require chainId or network.

[the remark this answers is not in the document]
Comment (KuphJr) [reply packages/composites/proof-of-reserves/src/utils/protocol.ts:32]: The problem is, sometimes they ARE optional.  For example, the 'eth-balance' indexer does NOT require chainId or network.

File: packages/composites/proof-of-reserves/src/utils/protocol.ts [hunk 6]
```diff
@@ -41,11 +54,30 @@ export const runProtocolAdapter = async (
   return callAdapter(execute, context, next, '_onProtocol')
 }
 
-const listAdapter = (jobRunID: string, data: { addresses: string[] }) => {
+const listAdapter = (
+  jobRunID: string,
+  data: { addresses: AddressList; chainId?: string; network?: string },
```
Comment (boxhock) [inline packages/composites/proof-of-reserves/src/utils/protocol.ts:59]: If using the suggested change above, we get:

```suggestion
  data: AddressList,
```

**Review by boxhock — CHANGES_REQUESTED** [review]

**Comment by boxhock** [comment]
This change was moved to #1896

**Pull request closed without merging**